drivers/serial/usart_gd32.c

```diff
@@ -93,25 +148,201 @@ static int usart_gd32_err_check(const struct device *dev)
 	return errors;
 }
 
+#ifdef CONFIG_UART_INTERRUPT_DRIVEN
+/** Interrupt driven FIFO fill function */
+int usart_gd32_fifo_fill(const struct device *dev, const uint8_t *tx_data,
+			int len)
+{
+	const struct gd32_usart_config *const cfg = dev->config;
+	uint8_t num_tx = 0U;
+
+	while ((len - num_tx > 0) &&
+		usart_flag_get(cfg->reg, USART_FLAG_TBE)) {
+		usart_data_transmit(cfg->reg, tx_data[num_tx++]);
+	}
+
+	return 0;
+}
+
+/** Interrupt driven FIFO read function */
+int usart_gd32_fifo_read(const struct device *dev, uint8_t *rx_data,
+			const int size)
```

Comment:
`const int size` should be vertically align with `const struct ...`

Reply:
Done!

Reply:
Done!